app/controllers/api/v1/users/devices_controller.rb

```diff
@@ -31,4 +25,12 @@ def show
     respond_with list
   end
 
+  private
+
+    def device_checkin_hash(device)
```

Comment:
This should be a method on the Device model, which the controller can just call

If a controller has a method based on an instance of a specific model, that method most likely should live in the model.